includes/plugins/woocommerce/my-account/class-my-account-ui-v1.php

```diff
@@ -119,5 +135,286 @@ public static function my_account_menu_items( $items ) {
 		unset( $items['customer-logout'] );
 		return $items;
 	}
+
+	/**
+	 * Remove required fields from the account settings form.
+	 *
+	 * @param array $required_fields The required fields.
+	 * @return array The required fields.
+	 */
+	public static function account_settings_required_fields( $required_fields ) {
+		unset( $required_fields['account_display_name'] );
+		return $required_fields;
+	}
+
+	/**
+	 * Display a series of modals to request account deletion.
+	 */
+	public static function delete_account_modal() {
+		// Only if the user is logged in and a reader.
+		if ( ! \is_user_logged_in() || ! Reader_Activation::is_user_reader( \wp_get_current_user() ) ) {
+			return;
+		}
+
+		// If the user has clicked the button from the delete account email, show the confirmation modal.
+		$delete_account_form = filter_input( INPUT_GET, WooCommerce_My_Account::DELETE_ACCOUNT_FORM, FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		if ( ! empty( $delete_account_form ) ) {
+			return self::delete_account_confirmation_modal();
+		}
+
+		ob_start();
+		?>
+		<h2 class="font-size newspack-ui__font--l">
+			<?php esc_html_e( 'Are you sure?', 'newspack-plugin' ); ?>
+		</h2>
+		<p>
+			<?php
+			esc_html_e( 'Deleting your account is permanent and cannot be undone. All your data will be removed from our systems. Your newsletter subscriptions and all recurring payments will be cancelled.', 'newspack-plugin' );
+			?>
+		</p>
+		<p>
+			<?php
+			esc_html_e( 'Instead of deleting your account, you may want to:', 'newspack-plugin' );
+			?>
+		</p>
+		<div class="newspack-ui__row">
+			<div>
+				<p class="font-size newspack-ui__font--s newspack-ui__font--bold"><?php esc_html_e( 'Subscriptions', 'newspack-plugin' ); ?></p>
+				<p class="newspack-ui__helper-text"><?php esc_html_e( 'Review and cancel active subscriptions.', 'newspack-plugin' ); ?></p>
+			</div>
+			<div class="newspack-ui__width--33">
+				<a class="newspack-ui__button newspack-ui__button--secondary newspack-ui__button--wide" href="<?php echo esc_url( \wc_get_endpoint_url( 'subscriptions', '', \wc_get_page_permalink( 'myaccount' ) ) ); ?>">
+					<?php esc_html_e( 'Manage subscriptions', 'newspack-plugin' ); ?>
+				</a>
+			</div>
+		</div>
```

Comment:
After testing again with a fresh reader account I realized that this is also rendering when I have no subscriptions, which points to an empty page with a weird notification markup:

<img width="886" alt="image" src="https://github.com/user-attachments/assets/b80f4758-c69e-43c0-940b-97ec2d8b1f4d" />

While at it, maybe also skip the "Manage newsletters" if the reader is not subscribed to any list?

Reply:
Good catch! a5e46dd2 tweaks this modal template so that the "Manage Subscriptions" and "Manage Newsletters" content will only be shown if the user has any active subscriptions for those entities. If not, the modal will look a lot simpler:

<img width="698" alt="Screenshot 2025-05-27 at 11 30 16 AM" src="https://github.com/user-attachments/assets/d72b8908-9622-4a26-83eb-e4e9cb020949" />